include/gsl/gsl_assert

```diff
@@ -72,16 +72,49 @@ struct fail_fast : public std::runtime_error
 {
     explicit fail_fast(char const* const message) : std::runtime_error(message) {}
 };
-}
+namespace details
+{
+template <typename Exception>
+[[noreturn]] void throw_exception(Exception&& exception);
```

Comment:
One final thing. Can we ensure that when the "no enforcement" macro is defined, `throw_exception` will still throw an exception? That won't affect `Ensures` or `Expects` but means the contract for `narrow()` (or any other functions that come in the future like it) can be maintained correctly.